packages/jbrowse-plugin-apollo/src/ApolloTextSearchAdapter/ApolloTextSearchAdapter.ts

```diff
@@ -0,0 +1,72 @@
+import { readConfObject } from '@jbrowse/core/configuration'
+import {
+  BaseAdapter,
+  BaseTextSearchAdapter,
+  BaseTextSearchArgs,
+} from '@jbrowse/core/data_adapters/BaseAdapter'
+import BaseResult from '@jbrowse/core/TextSearch/BaseResults'
+import { UriLocation } from '@jbrowse/core/util'
+import { getFetcher } from '@jbrowse/core/util/io'
+
+export class ApolloTextSearchAdapter
+  extends BaseAdapter
+  implements BaseTextSearchAdapter
+{
+  get baseURL() {
+    return readConfObject(this.config, 'baseURL').uri
+  }
+
+  get trackId() {
+    return readConfObject(this.config, 'trackId').uri
+  }
+
+  get internetAccountPreAuthorization():
+    | { authInfo: { token: string }; internetAccountType: string }
+    | undefined {
+    return readConfObject(this.config, 'baseURL')
+      .internetAccountPreAuthorization
+  }
+
+  async searchFeatures(term: string) {
+    const url = new URL(`features/searchFeatures?term=${term}`, this.baseURL)
+    const uri = url.toString()
+    const location: UriLocation = { locationType: 'UriLocation', uri }
+    if (this.internetAccountPreAuthorization) {
+      location.internetAccountPreAuthorization =
+        this.internetAccountPreAuthorization
+    }
+    const fetch = getFetcher(location, this.pluginManager)
+    return fetch(uri)
+      .then((res) => res.json())
+      .catch((err) => err)
```

Comment:
This should be in async-await style for consistency with existing code.
```suggestion
    const response = await fetch(uri)
    return response.json()
```